### PR TITLE
resgroup: avoid re-writing pid to cgroup if group is not changed

### PR DESCRIPTION
**Pull request by pengzhout**

### CODE REVIEW

File: src/backend/utils/resgroup/resgroup-ops-linux.c [hunk 1]
```diff
@@ -716,8 +717,14 @@ ResGroupOps_DestroyGroup(Oid group)
 void
 ResGroupOps_AssignGroup(Oid group, int pid)
 {
+	if (group == InvalidOid ||
```
Comment (pivotal-ning-yu) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: `group == InvalidOid` was supposed to move `pid` into the gpdb toplevel cgroup, we'd better keep this behavior.

File: src/backend/utils/resgroup/resgroup-ops-linux.c [hunk 2]
```diff
@@ -66,6 +66,7 @@ static void getCgMemoryInfo(uint64 *cgram, uint64 *cgmemsw);
 static int getOvercommitRatio(void);
 static void detectCgroupMountPoint(void);
 
+static int currentGroupIdInCGroup = InvalidOid;
```
Comment (pivotal-ning-yu) [inline src/backend/utils/resgroup/resgroup-ops-linux.c]: Better to use `Oid` as the type?

**Comment by pengzhout** [comment]
@pivotal-ning-yu  updated as your comment, thanks.

**Comment by pivotal-ning-yu** [comment]
Cool, LGTM.

**Comment by pengzhout** [comment]
merged

**Pull request merged**